godotenv.go

```diff
@@ -109,17 +109,54 @@ func Parse(r io.Reader) (envMap map[string]string, err error) {
 		return
 	}
 
+	var multilineKey, buffer string
+	multilineValue := false
 	for _, fullLine := range lines {
 		if !isIgnoredLine(fullLine) {
 			var key, value string
-			key, value, err = parseLine(fullLine, envMap)
+			var isMultilineValue bool
+			key, value, isMultilineValue, err = parseLine(fullLine, envMap, multilineValue)
 
 			if err != nil {
 				return
 			}
-			envMap[key] = value
+
+			if multilineValue == true && isMultilineValue == false && len(key) <= 0 {
```

Comment:
I have to admit I'm really confused by the two multilineValue booleans, and I'm not sure if that's a structural problem or a naming problem of the variables.

Reply:
Yeah; this is a fair comment. I'm not particularly happy with the implementation and thus wanted to get some feedback from you since is your codebase.

I'll close this PR and will refactor in a cleaner way probably this weekend.

🙇